staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -219,8 +223,8 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 		kubeconfigs = append(kubeconfigs, config)
 	}
 
-	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 {
-		klog.Warningf("Config not found: %s", strings.Join(missingList, ", "))
+	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 && rules.WarnIfAllMissingCallback != nil {
```

Comment:
when the caller does not use NewDefaultClientConfigLoadingRules(), and does not set rules.WarnIfAllMissingCallback explicitly, it would be nil. In that case, it should default to klog.Warningf, which is the previous behavior.

Reply:
I implemented this behavior by means of the unexported variable `ClientConfigLoadingRules.warningMuted`. In addition, the field `ClientConfigLoadingRules.warnIfAllMissingCallback` has been unexported and made customizable by using a setter method.